runtime/libpgmath/CMakeLists.txt

```diff
@@ -46,23 +46,50 @@ if (LIBPGMATH_STANDALONE_BUILD)
+                    "Libpgmath test suite will be disabled.")
     set(LLVM_INCLUDE_TESTS OFF)
   endif()
 endif()
 
-if(CMAKE_C_COMPILER_ID STREQUAL "GNU" AND ${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64|AMD64")
-  if(CMAKE_C_COMPILER_VERSION VERSION_LESS "7.2.0")
-    message(FATAL_ERROR "Found gcc at " ${CMAKE_C_COMPILER} " version " ${CMAKE_C_COMPILER_VERSION} ", but version 7.2.0 or newer required.")
+if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
+  if ("${LLVM_FLANG_CPU_TARGET}" STREQUAL "")
+    message(STATUS "Setting libpgmath CPU target to 'native'")
+    set(LLVM_FLANG_CPU_TARGET "native")
+  else()
+    message(STATUS "Libpgmath CPU target explicitly set to ${LLVM_FLANG_CPU_TARGET}")
   endif()
-# elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-#   if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "your.required.msvc.version")
-#     message(FATAL_ERROR "Insufficient msvc version")
-#   endif()
-# elseif(...)
-# # etc.
 endif()
 
+if(CMAKE_C_COMPILER_ID STREQUAL "GNU" AND ${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64|AMD64|aarch64")
+  if(CMAKE_C_COMPILER_VERSION VERSION_LESS "7.1.0")
+    message(FATAL_ERROR "Found gcc at " ${CMAKE_C_COMPILER} " version " ${CMAKE_C_COMPILER_VERSION} ", but version 7.1.0 or newer required.")
+  else()
+    message(STATUS "Found GCC Version ${CMAKE_C_COMPILER_VERSION}.")
+  endif()
+  # elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  #   if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "your.required.msvc.version")
+  #     message(FATAL_ERROR "Insufficient msvc version")
+  #   endif()
+  # elseif(...)
+  # # etc.
+endif()
+
+# if(CMAKE_C_COMPILER_ID STREQUAL "GNU" AND ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
+#   string(REPLACE "-O2" "-O3 -finline-functions -funroll-loops" CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
+#   string(REPLACE "-O2" "-O3 -finline-functions -funroll-loops" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+#   string(REPLACE "-std=c++11" "-std=gnu++11" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+#   string(REPLACE "-fno-tree-vectorize" "-ftree-vectorize" CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
+#   string(REPLACE "-fno-tree-vectorize" "-ftree-vectorize" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+#   string(REPLACE "-fno-tree-slp-vectorize" "-ftree-slp-vectorize" CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
+#   string(REPLACE "-fno-tree-slp-vectorize" "-ftree-slp-vectorize" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+#   string(REPLACE "-fno-reorder-blocks" "-freorder-blocks" CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
+#   string(REPLACE "-fno-reorder-blocks" "-freorder-blocks" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+#   string(REPLACE "-fno-toplevel-reorder" "-ftoplevel-reorder" CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
+#   string(REPLACE "-fno-toplevel-reorder" "-ftoplevel-reorder -felide-constructors" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+#   string(REPLACE "-std=c++11" "-std=gnu++11" CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG})
+#   string(REPLACE "-std=c++11" "-std=gnu++11" CMAKE_CXX_FLAGS_RELWITHDEBINFO ${CMAKE_CXX_FLAGS_RELWITHDEBINFO})
+# endif()
+
```

Comment:
AFAIK this can be un-commented and fixed.

Reply:
Yes, I already fixed this in my internal patch.

Reply:
Why does this patch require, for example, that -std=gnu++11 be used instead of c++11?